client/Packages/com.beamable.server/Editor/Usam/Editor/CodeService.cs

```diff
@@ -870,16 +871,34 @@ private async Promise CreateStorage(string storageName, List<IBeamoServiceDefini
 				deps[i] = additionalReferences[i].BeamoId;
 			}
 
+			Promise errorPromise = new Promise();
+			string errorMessage = string.Empty;
+
 			var storageArgs = new ProjectNewStorageArgs
 			{
 				name = service,
 				serviceDirectory = StandaloneMicroservicesPath,
 				sln = slnPath,
 				linkTo = deps,
 			};
-			var storageCommand = _cli.ProjectNewStorage(storageArgs);
+			var storageCommand = _cli.ProjectNewStorage(storageArgs).OnError((cb) =>
+			{
+				errorMessage = $"Error creating storage: {storageName}. Message=[{cb.data.message}] Stacktrace=[{cb.data.stackTrace}]";
+				errorPromise.CompleteSuccess();
+			});
 			await storageCommand.Run();
 
+			await RefreshServices();
+
+			var definition = ServiceDefinitions.FirstOrDefault(def => def.BeamoId == storageName);
+
+			if (definition == null)
+			{
+				await errorPromise;

```

Comment:
I'm not really a fan of this promise thing, if something went wrong with the creation but it did create a project anyway, this wouldn't caught that. But the problem with making this better, is that we are starting a new thread for calling the `OnError` stream event, which makes throwing the exception there never getting caught in the creation promise. Is there another way to handle this? I'm not sure if we could just not start new threads for these callbacks, but that would definitely fix this.